flink-runtime/src/main/java/org/apache/flink/runtime/checkpoint/CheckpointCoordinator.java

```diff
@@ -296,15 +298,42 @@ public boolean isShutdown() {
 		checkNotNull(targetDirectory, "Savepoint target directory");
 
 		CheckpointProperties props = CheckpointProperties.forStandardSavepoint();
-		CheckpointTriggerResult result = triggerCheckpoint(timestamp, props, targetDirectory, false);
 
-		if (result.isSuccess()) {
-			return result.getPendingCheckpoint().getCompletionFuture();
-		}
-		else {
-			Throwable cause = new Exception("Failed to trigger savepoint: " + result.getFailureReason().message());
-			return FlinkCompletableFuture.completedExceptionally(cause);
+		// Create the unique savepoint directory
+		final String savepointDirectory = SavepointStore
```

Comment:
We could also not create the directory here, but instead simply create the path name and send it to the tasks.

@StephanEwen Maybe creating directories like these is something we should keep in mind for the state backends on the JobManager?